riscv-executor/src/lib.rs

```diff
@@ -200,7 +183,7 @@ pub struct RegWrite<F: FieldElement> {
     row: usize,
     /// Index of the register in the register bank.
     reg_idx: u16,
-    val: Elem<F>,
+    val: F,
```

Comment:
I think this might not be ideal for performance. Most of the register operations are actually over u32/i64, and here we're forcing it to be always F.

Reply:
Or maybe I'm tripping, but wanted to bring it up anyway.

Reply:
General register operations during execution are still over Elem. This just converts `i64 -> F` when storing the reg write operation in `Trace` mode. 
Note that these are only for the actual "asm write registers" (`pc` and `query_arg_*`). I think this won't make much difference but I can double check.

Reply:
Ah right, makes sense then.